numba_cuda/numba/cuda/tests/test_analysis.py

```diff
@@ -0,0 +1,1109 @@
+                )
+
+    @skip_on_cudasim("Analysis passes not done in simulator")
+    def test_single_if_else_global(self):
+        global c_test_single_if_else_global
+
+        for c_inp, prune in (self._TRUTHY, False), (self._FALSEY, True):
+            for c in c_inp:
+                c_test_single_if_else_global = c
+
+                def func(x):
+                    if c_test_single_if_else_global:
+                        return 3.14159, c_test_single_if_else_global
+                    else:
+                        return 1.61803, c_test_single_if_else_global
+
+                self.assert_prune(
+                    func, (types.NoneType("none"),), [prune], None
+                )
+
+    @skip_on_cudasim("Analysis passes not done in simulator")
+    def test_single_if_else_negate_global(self):
+        global c_test_single_if_else_negate_global
+
+        for c_inp, prune in (self._TRUTHY, False), (self._FALSEY, True):
+            for c in c_inp:
+                c_test_single_if_else_negate_global = c
+
+                def func(x):
+                    if not c_test_single_if_else_negate_global:
+                        return 3.14159, c_test_single_if_else_negate_global
+                    else:
+                        return 1.61803, c_test_single_if_else_negate_global
+
+                self.assert_prune(
+                    func, (types.NoneType("none"),), [prune], None
+                )
+
+    @skip_on_cudasim("Analysis passes not done in simulator")
+    def test_issue_5618(self):
+        @njit
+        def foo():
+            values = np.zeros(1)
+            tmp = 666
+            if tmp:
+                values[0] = tmp
+            return values
+
+        self.assertPreciseEqual(foo.py_func()[0], 666.0)
+        self.assertPreciseEqual(foo()[0], 666.0)
+
+
+class TestBranchPruneSSA(MemoryLeakMixin, TestCase):
+    # Tests SSA rewiring of phi nodes after branch pruning.
+
+    class SSAPrunerCompiler(CompilerBase):
+        def define_pipelines(self):
+            # This is a simple pipeline that does branch pruning on IR in SSA
+            # form, then types and lowers as per the standard nopython pipeline.
+            pm = PassManager("testing pm")
+            pm.add_pass(TranslateByteCode, "analyzing bytecode")
+            pm.add_pass(IRProcessing, "processing IR")
+            # SSA early
+            pm.add_pass(ReconstructSSA, "ssa")
+            pm.add_pass(DeadBranchPrune, "dead branch pruning")
+            # type and then lower as usual
+            pm.add_pass(PreserveIR, "preserves the IR as metadata")
+            dpb = DefaultPassBuilder
+            typed_passes = dpb.define_typed_pipeline(self.state)
+            pm.passes.extend(typed_passes.passes)
+            lowering_passes = dpb.define_nopython_lowering_pipeline(self.state)
+            pm.passes.extend(lowering_passes.passes)
+            pm.finalize()
+            return [pm]
+
+
```

Comment:
This class doesn't seem to be doing anything - were there some tests in it that are now removed?